src/v/archival/tests/ntp_archiver_test.cc

```diff
@@ -550,38 +551,70 @@ SEASTAR_THREAD_TEST_CASE(test_archival_policy_timeboxed_uploads) {
 
 namespace archival::internal {
 
-// A replacement for NTP archiver which does no operations. Can be placed in the
-// scheduler service to make sure that the service fixture does not interfere
-// with another archiver started explicitly in a unit test.
-class no_op_archiver final : public archival::ntp_archiver {
-public:
-    ss::future<ntp_archiver::batch_result> upload_next_candidates(
-      std::optional<model::offset> last_stable_offset_override) override {
-        ntp_archiver::batch_result result{
-          .non_compacted_upload_result = {}, .compacted_upload_result = {}};
-        return ss::make_ready_future<ntp_archiver::batch_result>(result);
-    }
-};
-
+/// Replaces the ntp archiver in scheduler with an object which has a remote
+/// connection configured to a port which is not being listened to. The new
+/// archiver does not make any progress, freeing the unit test to create an
+/// archiver of its own and performing actions and making assertions about them
+/// in a clean environment.
+///
+/// We initialize a remote because the newly created archiver expects a
+/// reference to a remote which should remain alive for the duration of the
+/// test. The accessor must be closed before the test ends.
 class scheduler_service_accessor {
```

Comment:
nit: maybe this class should have a different name now? It's only use is to replace a current archiver, which may be confusing if someone tries to extend it for some other normal usage.

Reply:
sure, changed to `archiver_replacer`